### PR TITLE
Travis improvements (folder structure, mac support etc.)

### PR DESCRIPTION
**Pull request by olehermanse**
Based on #2770

### CODE REVIEW

**Comment by nickanderson** [comment]
@tzz  you might be interested in this PR

**Comment by tzz** [comment]
Very cool, thank you for working on this!

**Comment by jimis** [comment]
Very cool indeed! Will look into it after 3.10.1.

**Comment by olehermanse** [comment]
@jimis Review when you have time. Ready AFAIK. I think it makes sense to merge with just compile only on mac, since running tests have proven a little difficult, and this PR includes a decent amount of changes already.

**Comment by jimis** [comment]
I agree! Will review and merge right after 3.10.1.

**Comment by olehermanse** [comment]
@jimis Please take a look at this one as well. Now that release is done.

File: .travis.yml [hunk 1]
```diff
@@ -8,6 +7,7 @@ addons:
 
 env:
   global:
+    - CI_NODE_TOTAL=6
```
Comment (jimis) [inline .travis.yml]: Comment on why

File: .travis.yml [hunk 2]
```diff
@@ -20,104 +20,59 @@ env:
...
 deploy:
   skip_cleanup: true
   provider: releases
   # Remember to set it as private variable in Travis-CI settings
   api_key: $GITHUB_RELEASE_TOKEN
   # Does not work, I get the error that "true" is not a boolean
-  #  prerelease: $IS_PRERELEASE
-  file:
-    - "$DIST_TARBALL"
+  prerelease: $IS_PRERELEASE
```
Comment (jimis) [inline .travis.yml]: This was a bug in Travis-CI. It works now so you uncommented it? Then the comment should be removed.

File: libcompat/strchrnul.c [hunk 3]
```diff
@@ -12,7 +12,7 @@ char *strchrnul(const char *s, int c)
 
     if (p == NULL)
     {
-        return s + strlen(s);
```
Comment (jimis) [inline libcompat/strchrnul.c]: why?

**Pull request merged**